### ISSUE TITLE
OCS: WebDAV access should not contain folder name

### ISSUE THREAD

**Issue by LukasReschke** [pydio/pydio-core#1066]
When sharing a folder with ownCloud we're using `public.php/webdav/`. The change root into the specific folder is performed by ownCloud itself.

When sharing a folder `awesome` with Pydio it requests `/public.php/webdav/awesome` instead of `/public.php/webdav` => :bomb: 

I'll create a PR for that so that it works for ownCloud.

**Comment by cdujeu** [comment]
@LukasReschke I actually kind of hacked our own system to be consistent with the API :-)
Initially, for us, when we share a folder the "natural" endpoint was /dav/ as well, but when we share a file it would be /dav/filename.txt. 
I changed this to make it more consistent, and because the API does not convey the information whether we are sharing a file or a folder.

**Comment by LukasReschke** [comment]
At the moment we're detecting this in ownCloud by looking whether the top resource is not a collection. If you look at the responses to a PROPFIND on `/master/public.php/webdav/` you will see that the folders contain a `d:collection` on the `resourcetype` while the single shared file is not marked as such.

Would it be possible to adjust that in Pydio as well like that? Changing the endpoint on our side would most likely break existing shares.

Shared folder (empty):
```xml
<?xml version="1.0"?>
<d:multistatus xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns" xmlns:oc="http://owncloud.org/ns">
  <d:response>
    <d:href>/master/public.php/webdav/</d:href>
    <d:propstat>
      <d:prop>
        <d:getlastmodified>Tue, 01 Mar 2016 07:39:25 GMT</d:getlastmodified>
        <d:resourcetype>
          <d:collection/>
        </d:resourcetype>
        <d:quota-used-bytes>0</d:quota-used-bytes>
        <d:quota-available-bytes>26714128384</d:quota-available-bytes>
        <d:getetag>&quot;56d5472d22166&quot;</d:getetag>
      </d:prop>
      <d:status>HTTP/1.1 200 OK</d:status>
    </d:propstat>
  </d:response>
</d:multistatus>
```
Shared folder (non empty):
```xml
<?xml version="1.0"?>
<d:multistatus xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns" xmlns:oc="http://owncloud.org/ns">
  <d:response>
    <d:href>/master/public.php/webdav/</d:href>
    <d:propstat>
      <d:prop>
        <d:getlastmodified>Tue, 01 Mar 2016 07:38:28 GMT</d:getlastmodified>
        <d:resourcetype>
          <d:collection/>
        </d:resourcetype>
        <d:quota-used-bytes>2197</d:quota-used-bytes>
        <d:quota-available-bytes>26715156480</d:quota-available-bytes>
        <d:getetag>&quot;56d546f4e3124&quot;</d:getetag>
      </d:prop>
      <d:status>HTTP/1.1 200 OK</d:status>
    </d:propstat>
  </d:response>
  <d:response>
    <d:href>/master/public.php/webdav/insideFile.jpg</d:href>
    <d:propstat>
      <d:prop>
        <d:getlastmodified>Tue, 01 Mar 2016 07:37:11 GMT</d:getlastmodified>
        <d:getcontentlength>2197</d:getcontentlength>
        <d:resourcetype/>
        <d:getetag>&quot;9f692d3943587bd6df049a2f57c50979&quot;</d:getetag>
        <d:getcontenttype>image/jpeg</d:getcontenttype>
      </d:prop>
      <d:status>HTTP/1.1 200 OK</d:status>
    </d:propstat>
  </d:response>
</d:multistatus>
```

Shared file:
```xml
<?xml version="1.0"?>
<d:multistatus xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns" xmlns:oc="http://owncloud.org/ns">
  <d:response>
    <d:href>/master/public.php/webdav/</d:href>
    <d:propstat>
      <d:prop>
        <d:getlastmodified>Tue, 01 Mar 2016 06:02:36 GMT</d:getlastmodified>
        <d:getcontentlength>163</d:getcontentlength>
        <d:resourcetype/>
        <d:getetag>&quot;bbdf083e23e0af725bf45bcdf47d591d&quot;</d:getetag>
        <d:getcontenttype>text/plain</d:getcontenttype>
      </d:prop>
      <d:status>HTTP/1.1 200 OK</d:status>
    </d:propstat>
  </d:response>
</d:multistatus>
```

**Comment by cdujeu** [comment]
... too bad I spent some time on that already ...

So, for a **shared file** you also point to /dav/, not /dav/DocumentName.txt right? Or both would return the same answer? 
-c

**Comment by LukasReschke** [comment]
> ... too bad I spent some time on that already ...

Sorry. But keep your code in the Git history. We can happily talk about using it in a future and improved API :-)

> So, for a shared file you also point to /dav/, not /dav/DocumentName.txt right? Or both would return the same answer? 

We always point to DAV. Correct. What we do is a PROPFIND on the resource and determine whether it's a file or a folder.

If it is a file we just do a GET on `/dav/` (it will return the file directly), if it is a folder we do a GET on `/dav/filename.txt`.

**Comment by cdujeu** [comment]
Fixed in d71a66eae40d7c590749c19df89036ebadc6121b
